<launch>

  <arg name="robot_model_1"                     default="px100"/>
  <arg name="robot_name_1"                      default="arm_1"/>
  <arg name="base_link_1"                       default="base_link"/>
  <arg name="modes_1"                           default="$(find-pkg-share control)/config/modes_1.yaml"/>

  <arg name="robot_model_2"                     default="px100"/>
  <arg name="robot_name_2"                      default="arm_2"/>
  <arg name="base_link_2"                       default="base_link"/>
  <arg name="modes_2"                           default="$(find-pkg-share control)/config/modes_2.yaml"/>

  <arg name="use_sim"                           default="false"/>
  <arg name="use_dual_rviz"                     default="false"/>
  <arg name="rvizconfig"                        default="$(find-pkg-share control)/rviz/xsarm_dual.rviz" />

  <include file="$(find-pkg-share interbotix_xsarm_control)/launch/xsarm_control.launch.py">
    <arg name="robot_model"                       value="$(var robot_model_1)"/>
    <arg name="robot_name"                        value="$(var robot_name_1)"/>
    <arg name="base_link_frame"                   value="$(var base_link_1)"/>
    <arg name="use_world_frame"                   value="false"/>
    <arg name="use_rviz"                          value="false"/>
    <arg name="mode_configs"                      value="$(var modes_1)"/>
    <arg name="use_sim"                           value="$(var use_sim)"/>
  </include>

  <include file="$(find-pkg-share interbotix_xsarm_control)/launch/xsarm_control.launch.py">
    <arg name="robot_model"                       value="$(var robot_model_2)"/>
    <arg name="robot_name"                        value="$(var robot_name_2)"/>
    <arg name="base_link_frame"                   value="$(var base_link_2)"/>
    <arg name="use_world_frame"                   value="false"/>
    <arg name="use_rviz"                          value="false"/>
    <arg name="mode_configs"                      value="$(var modes_2)"/>
    <arg name="use_sim"                           value="$(var use_sim)"/>
  </include>

  <node
    name="robot_1_transform_broadcaster"
    pkg="tf2_ros"
    exec="static_transform_publisher"
    args="0 -0.25 0 0 0 0 /world /$(var robot_name_1)/$(var base_link_1)"/>

  <node
    name="robot_2_transform_broadcaster"
    pkg="tf2_ros"
    exec="static_transform_publisher"
    args="0 0.25 0 0 0 0 /world /$(var robot_name_2)/$(var base_link_2)"/>

  <node 
    name="rviz2"
    pkg="rviz2"
    exec="rviz2"
    args="-d $(var rvizconfig)"
    if="$(eval '\'$(var use_dual_rviz)\' == \'true\'')"/>

  <node pkg="control" exec="robot_control" name="robot_control"/>

</launch>
